itest/rust/src/builtin_tests/string/string_name_test.rs

```diff
@@ -127,15 +127,17 @@ fn string_name_is_empty() {
 
 #[itest]
 #[cfg(since_api = "4.2")]
-fn string_name_from_latin1_with_nul() {
-    let cases: [(&[u8], &str); 3] = [
-        (b"pure ASCII\t[~]\0", "pure ASCII\t[~]\0"),
-        (b"\xB1\0", "±"),
-        (b"Latin-1 \xA3 \xB1 text \xBE\0", "Latin-1 £ ± text ¾"),
+fn string_name_from_cstr() {
+    use std::ffi::CStr;
+
+    let cases: [(&CStr, &str); 3] = [
+        (c"pure ASCII\t[~]", "pure ASCII\t[~]\0"),
```

Comment:
I just noticed now that the right-hand-side here has an explicit `\0` character at the end.

Seems like the constructor taking UTF-8 strings skips that, as the `assert_eq` below passes? Could possibly warrant a separate test 🤔

Reply:
oh huh yeah

Reply:
apparently any bytes after a null byte are ignored? ig they must treat it as a c-string somewhere

Reply:
Maybe we can document that on `String`/`StringName`? Iterating through the entire string to check for internal NULs might be overkill, given that it's extremely unlikely to happen in practice...

Reply:
It'd be nice to have a more central place to provide more general documentation of the string types. Currently all the builtins are in `godot::builtin` but maybe having `godot::builtin::string` could be nice so that we can write a more general overview of all the string types in there.

They'll all still be in the prelude so it shouldn't cause an issue with making it harder to access them for most people at least.